addons/base_import/models.py

```diff
@@ -126,6 +134,56 @@ def get_fields(self, cr, uid, model, context=None,
         # TODO: cache on model?
         return fields
 
+    def _read_import_file(self, record, options):
+        rows = []
+        if record.file_type == 'text/csv':
+            rows = self._read_csv(record, options)
+        elif record.file_type == 'application/vnd.ms-excel' or record.file_type == 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet':
+            if xlrd is None:
+                raise UserError(_(" XLRD library not found. XLS/XLSX file would not import."))
+            else:
+                rows = self._read_xls_xlsx(record, options)
+        elif record.file_type == 'application/vnd.oasis.opendocument.spreadsheet':
+            rows = self._read_ods(record, options)
+        return rows
+
+    def _read_xls_xlsx(self, record, options):
+        book = xlrd.open_workbook(file_contents=record.file)
+        bk = book.sheet_by_index(0)
+        keys = []
+        for row_index in range(0, bk.nrows):
+            keys.append([])
+            for col_index in range(bk.ncols):
+                if bk.cell_type(row_index, col_index) == 3:
+                    date_decimal = str(bk.cell_value(row_index, col_index)-int(bk.cell_value(row_index, col_index)))[1:].split('.')[1]
+                    if date_decimal == '0':
+                        keys[-1].append(datetime.datetime(* \
+                        (xlrd.xldate_as_tuple(bk.cell_value(row_index, col_index), \
+                            book.datemode))).strftime('%Y-%m-%d'))
+                    else:
+                        keys[-1].append(datetime.datetime(* \
+                        (xlrd.xldate_as_tuple(bk.cell_value(row_index, col_index), \
+                            book.datemode))).strftime('%Y-%m-%d %H:%M:%S'))
+                elif bk.cell_type(row_index, col_index) == 2:
+                    keys[-1].append(str(bk.cell_value(row_index, col_index)))
+                elif bk.cell_type(row_index, col_index) == 4:
+                    if bk.cell_value(row_index, col_index) == 1:
+                        keys[-1].append('True')
+                    elif bk.cell_value(row_index, col_index) == 0:
+                        keys[-1].append('False')
+                else:
+                    keys[-1].append(bk.cell_value(row_index, col_index))
+        encoding = options.get('encoding', 'utf-8')
+        dict_list = itertools.imap(lambda row: [item.decode(encoding) for item in row], keys)
```

Comment:
The data is already decoded, this step makes no sense: store unicode values in the first place (so don't reencode text cells badly)